### DISCUSSION TITLE
Clingon CLI - Dev tool

### DISCUSSION THREAD

**Discussion by ipetinate** [frontendbr/forum#2489] [Mostre seu trabalho]
Eai pessoal, tudo bem?

Queria mostrar um pouco de um projeto que venho trabalhando recentemente: uma ferramenta para desenvolvedores.

Com o Clingon, você pode criar seus componentes, páginas, funções (e futuramente) muito mais, com alguns comandos, gere tudo que seu projeto precisa: componentes, testes, stories, se precisar por tudo em uma pasta você consegue, pode configurar detalhes de alias, export, etc.

É uma ferramenta open source, caso queira interagir no [Discussions](https://github.com/ipetinate/clingon/discussions) ou abrir uma [Issue](https://github.com/ipetinate/clingon/issues) informando um bug, fique a vontade.

Temos poucas funcionalidades no momento, e geramos arquivos para Vue e React, mas a ideia é expandir, ter templates opinados para Solid, Svelte, Astro, e outros frameworks, além de que em breve você vai poder criar Classes, Types, Interfaces, Enums, além dos tipos disponíveis atualmente.
E uma das próximas funcionalidades a ser desenvolvida é a possibilidade de usar templates locais, onde você poderá criar qualquer tipo de arquivo ou recurso usando a ferramenta.

Ao longo do tempo ela vai ganhar novas funcionalidades, espero que você a use e faça parte desse amadurecimento.

####  Site Oficial

https://clingon.dev

#### Repositório

https://github.com/ipetinate/clingon

#### NPM

https://npmjs.org/package/clingon

---

![clingon-wallpaper](https://github.com/frontendbr/forum/assets/15758789/34db45b0-ed59-48e6-ae11-4871b35d5f43)

**Comment by ghost** [comment]
O nome tem haver com star trek?

**Comment by ipetinate** [comment]
> O nome tem haver com star trek?

Opa, com certeza hehe 🖖
Eu tinha chamado de Tricorder mas já tinha um pacote NPM com esse nome, aí mudei para esse.

Na doc oficial eu falo sobre isso HTTPS://clingon.dev/about

**Comment by ipetinate** [comment]
Acabei de lançar a v1 da ferramenta, com novos recursos, melhorias e correções importantes.

👉 https://www.clingon.dev/

Novo e principal recurso: Modo avançado, permite o uso de templates locais e criação de qualquer tipo de arquivo, com qualquer tipo de implementação e extensão.
  - 👉 https://www.clingon.dev/docs/advanced

**Reply by ghost** [reply]
que boa nova!
